drivers/src/idevid_csr.rs

```diff
@@ -0,0 +1,49 @@
+// Licensed under the Apache-2.0 license
+
+use core::mem::size_of;
+use zerocopy::{AsBytes, FromBytes};
+use zeroize::Zeroize;
+
+use crate::memory_layout;
+
+pub const MAX_CSR_SIZE: usize = 512;
+
+#[derive(FromBytes, AsBytes, Zeroize)]
+#[repr(C)]
+pub struct IDevIDCsr {
+    pub csr: [u8; MAX_CSR_SIZE],
+    pub csr_len: u32,
```

Comment:
I'm not positive, but it might be better for alignment to put `csr_len` as the first field.

Reply:
I think so too (at least similar structs do). I've made this change and will include it in the next rev.